src/io/flutter/jxbrowser/JxBrowserManager.java

```diff
@@ -0,0 +1,227 @@
+          loadClasses(files);
+          return;
+        }
+      }
+      // If jxbrowser was not downloaded within allowed time.
+      LOG.info(project.getName() + ": JxBrowser download timed out");
+      Notifications.Bus.notify(
+        new Notification(
+          FlutterMessages.FLUTTER_NOTIFICATION_GROUP_ID,
+          FlutterIcons.Flutter,
+          "JxBrowser download timed out",
+          null,
+          // TODO(helin24): Replace with better message or retry download action.
+          String.format("Retry by deleting files in %s and restarting IntelliJ", DOWNLOAD_PATH),
+          NotificationType.INFORMATION,
+          null
+        )
+      );
+      status.set(JxBrowserStatus.INSTALLATION_FAILED);
+      return;
+    }
+
+    // Delete any already existing files.
+    // TODO: Handle if files cannot be deleted.
+    for (File file : files) {
+      if (file.exists()) {
+        if (!file.delete()) {
+          LOG.info(project.getName() + ": Existing file could not be deleted - " + file.getAbsolutePath());
+        }
+      }
+    }
+
+    downloadJxBrowser(project, fileNames, tempLoadingFile);
+  }
+
+  private void downloadJxBrowser(Project project, String[] fileNames, File tempLoadingFile) {
+    final List<FileDownloader> fileDownloaders = new ArrayList<>();
+    final DownloadableFileService service = DownloadableFileService.getInstance();
+    for (String fileName : fileNames) {
+      final DownloadableFileDescription
+        description = service.createFileDescription(JxBrowserUtils.getDistributionLink(fileName), fileName);
+      fileDownloaders.add(service.createDownloader(Collections.singletonList(description), fileName));
+    }
+
+    final Task.Backgroundable task = new Task.Backgroundable(project, "Downloading jxbrowser") {
+      @Override
+      public void run(@NotNull ProgressIndicator indicator) {
+        String currentFileName = null;
+        try {
+          final List<File> files = new ArrayList<>();
+          for (int i = 0; i < fileDownloaders.size(); i++) {
+            final FileDownloader downloader = fileDownloaders.get(i);
+            currentFileName = fileNames[i];
+            final Pair<File, DownloadableFileDescription> download =
+              ContainerUtil.getFirstItem(downloader.download(new File(DOWNLOAD_PATH)));
+            final File file = download != null ? download.first : null;
+            if (file != null) {
+              files.add(file);
+              LOG.info(project.getName() + ": JxBrowser file downloaded: " + file.getAbsolutePath());
+            }
+          }
+
+          tempLoadingFile.delete();
+          loadClasses(files);
+        }
+        catch (IOException e) {
+          LOG.info(project.getName() + ": JxBrowser file downloaded failed: " + currentFileName);
+          e.printStackTrace();
+          status.set(JxBrowserStatus.INSTALLATION_FAILED);
+          tempLoadingFile.delete();
```

Comment:
keep in mind if the IDE is killed you may not get a chance to delete this file. Likely a file lock that is released if the IDE is killed or similar would be more robust.

Reply:
Removing the `tempLoadingFile` since we will only try once on initial startup of an IDE. We can let users manually try another download if they open up a devtools panel.